src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,111 @@
+     * "deadline S by T", "event S at T", where X is an integer, S is a string descriptor of the task, W is a
+     * singular keyword and T is a string descriptor of the date(s) and/or time(s) associated with the task.
+     *
+     * @return command from input string if input is of a valid format, "" otherwise
+     */
+    public static String parse(String input) {
+        input = input.trim();
+        String command = input.replaceAll(" .*", "");
+
+        input = input.trim();
+        if (input.equals("bye") || input.equals("list")) {
+            return command;
+        }
+
+        // Handle do, undo, delete
+        String firstWord = input.replaceAll(" .*", "");
+        if (firstWord.equals("do") || firstWord.equals("undo") || firstWord.equals("delete")) {
+            input = input.replaceAll(".* ", "");
+            if (input.matches("[0-9]+")) {
+                return command;
+            }
+            System.out.println("You need to specify the task you want to "+ firstWord + " by its index :c");
+            return "";
+        }
+
+        // Handle todo
+        if (firstWord.equals("todo")) {
+            input = input.substring(4).trim();
+            if (input.equals("")) {
+                System.out.println("Oops, you need to mention what the task is :c");
+                return "";
+            }
+            return command;
+        }
+
+        // Handle deadline
+        if (firstWord.equals("deadline")) {
+            input = input.substring(8).trim();
+            if (!input.contains(" by ")) {
+                System.out.println("Oops, you need to format deadline tasks as \"deadline X by Y\" :c");
+                return "";
+            }
+            String lastWord = input.substring(input.lastIndexOf(" ") + 1);
+            if (lastWord.equals("by")) {
+                return "";
+            }
+            return command;
+        }
+
+        // Handle event
+        if (firstWord.equals("event")) {
+            input = input.substring(5).trim();
+            if (!input.contains(" at ")) {
+                System.out.println("Oops, you need to format event tasks as \"event X at Y\" :c");
+                return "";
+            }
+            String lastWord = input.substring(input.lastIndexOf(" ")+1);
+            if (lastWord.equals("at")) {
+                return "";
+            }
+            return command;
+        }
+
+        // Handle find
+        if (firstWord.equals("find")) {
+            input = input.substring(4).trim();
+            if (input.equals("")) {
+                System.out.println("Oops, you need to mention what the keyword is :c");
+                return "";
+            }
+            if (input.contains(" ")) {
+                System.out.println("Oops, you can only search for one keyword at a time :c");
+                return "";
+            }
+            return command;
+        }
+
+        return "";
+    }
```

Comment:
How about making these logic sentence into a switch case statement? So that it is easier to follow and read!

Reply:
Thanks for highlighting this; I have now changed the code to make use of switch-case statements to make it more readable.